### DISCUSSION TITLE
PMD 7.0-rc1 issues with Ant

### DISCUSSION THREAD

**Discussion by road723** [pmd/pmd#4437] [Q&A] [answered]
Been messing around with PMD 7 and trying to get everything configured. Stuck on an issue with our Ant configuration  and getting the following error: The following language is not supported: `<sourceLanguage name="java" version="11" />`.

Perhaps I should wait until rc2 when the documentation is fully updated, but seems like this is an odd spot for a failure. I grabbed the new pmd-ant jar and the taskdef and classpath all seem to be setup similarly to how it was done in 6.x so not quite sure what else to look at just yet.

**Comment by jsotuyod** [comment]
First of, thanks for trying out the RC! It really helps. As for your error… that's unexpected… can you please double check that you have all 3 dependencies for PMD in place? `pmd-core`, `pmd-ant`, and `pmd-java`? The only reason I can think for that to happen is that `pmd-java` is missing…

**Reply by road723** [reply]
Indeed all 3 dependencies are in place, I've included my full target below. Only thing I have noticed is if I remove the sourceLanguage attribute line I end up with a java.lang.NoClassDefFoundError: org/slf4j/bridge/SLF4JBridgeHandler error so perhaps I am somehow missing something else?

```
<target name="pmdHtml">
	<path id="pmd.classpath">
		<fileset dir="${codeQuality.dir}" includes="*.jar"/>
		<fileset dir="${lib.dir}" includes="*.jar"/>
	</path>

	<taskdef name="pmd" classname="net.sourceforge.pmd.ant.PMDTask" classpathref="pmd.classpath"/>
	
	<path id="codeQuality.auxClasspath">
			<fileset dir="${lib.dir}" includes="*.jar"/>
		    <fileset dir="${codeQuality.dir}" includes="*.jar"/>
		    <fileset dir="${dist.dir}" includes="*.jar"/>
	</path>
	
	<pmd rulesetfiles="${pmd.dir}\ptsPmdConfig.xml" cacheLocation="${pmdCache.dir}\pmd.cache">
		<sourceLanguage name="java" version="11"/>
		<auxClasspath refid="codeQuality.auxClasspath" />
		<formatter type="xml" toFile="${pmd.dir}\ptsPmdReport.xml"/>
		<fileset dir="src">
			<include name="**/*.java"/>
		</fileset>
	</pmd>
	<xslt in="${pmd.dir}\ptsPmdReport.xml" style="${pmd.dir}\pmd-nicerhtml.xsl" out="${pmd.dir}\ptsPmdReport.html" />
		
</target>
```

**Comment by road723** [comment, marked as answer]
Looks like I found my issue as my Ivy task to pull the JARs was not configured correctly to pull everything I needed. PMD is running now and I look forward to 7!

**Reply by jsotuyod** [reply]
Glad to hear that!
